pkg/logs/processor/processor.go

```diff
@@ -104,27 +126,83 @@ func (p *Processor) run() {
 
 	for {
 		select {
+		// Processing, usual main loop
+		// ---------------------------
+
 		case msg, ok := <-p.inputChan:
 			if !ok { // channel has been closed
 				return
 			}
-			p.processMessage(msg)
+
+			// if we have to wait for an SDS configuration to start processing & forwarding
+			// the logs, that's here that we buffer the message
+			if p.sds.buffering {
+				// buffer until we receive a configuration
+				p.bufferMessage(msg)
+			} else {
+				// process the message
+				p.processMessage(msg)
+			}
+
 			p.mu.Lock() // block here if we're trying to flush synchronously
 			//nolint:staticcheck
 			p.mu.Unlock()
+
+		// SDS reconfiguration
+		// -------------------
+
 		case order := <-p.ReconfigChan:
 			p.mu.Lock()
-			if err := p.sds.Reconfigure(order); err != nil {
-				log.Errorf("Error while reconfiguring the SDS scanner: %v", err)
-				order.ResponseChan <- err
-			} else {
-				order.ResponseChan <- nil
-			}
+			p.applySDSReconfiguration(order)
 			p.mu.Unlock()
 		}
 	}
 }
 
+func (p *Processor) applySDSReconfiguration(order sds.ReconfigureOrder) {
+	isActive, err := p.sds.scanner.Reconfigure(order)
+	response := sds.ReconfigureResponse{
+		IsActive: isActive,
+		Err:      err,
+	}
+
+	if err != nil {
+		log.Errorf("Error while reconfiguring the SDS scanner: %v", err)
+	} else {
+		// no error while reconfiguring the SDS scanner? we should be
+		// ready starting the SDS processing if it was an Agent config
+		// or not ready anymore if after having reconfigured
+		// the SDS scanner it is not active anymore.
+		if isActive {
+			if p.sds.waitForConfig {
+				log.Debug("Processor ready with an SDS configuration.")
+				p.sds.buffering = false
+
+				// drain the buffer of messages if anything's in there
+				log.Debug("SDS: draining", len(p.sds.buffer), "buffered messages")
+				for _, msg := range p.sds.buffer {
+					p.processMessage(msg)
+				}
+
+				p.sds.buffer = p.sds.buffer[:0]
```

Comment:
I may be missing something, but we won't need the buffer ever again, right?
```suggestion
				p.sds.buffer = nil
```

Reply:
Oh thank you you're right it's way better that way, I have this in a stashed change somewhere 😅